src/server/pkg/pachhash/hash.go

```diff
@@ -0,0 +1,34 @@
+package pachhash
+
+import (
+	"encoding/hex"
+	"hash"
+
+	"golang.org/x/crypto/blake2b"
+)
+
+// OutputSize is the size of an Output in bytes
+const OutputSize = 32
+
+// Output is the output of the hash function.
+// Sum returns an Output
+type Output = [OutputSize]byte
+
+// New creates a new hasher.
+func New() hash.Hash {
+	h, err := blake2b.New256(nil)
+	if err != nil {
+		panic(err)
+	}
+	return h
+}
+
+// Sum computes a hash sum for a set of bytes.
+func Sum(data []byte) Output {
```

Comment:
It seems like it would be better to return a byte slice here, and just handle grabbing the 32 bytes under the hood. Is there a reason why we shouldn't do that?

Reply:
Yeah there are a few reasons.
1. This can prevent allocations, if output is just copied to its destination.
2. It can be used in map and sync.Map. `[]byte` cannot.  We are making use of this for the in memory cache.
3. It is a correctly sized hash, no chance of getting a short buffer if you expect this type.
4. Returning `[]byte` is a lowest common interface thing for `hash.Hash` you'll notice most of the cryptographic hash implementations include a `Sum256` or `Sum512`.